M2.el

```diff
@@ -354,30 +354,31 @@ can be executed with \\[M2-send-to-program]."
     ;; error messages, e.g.,
     ;; i1 : load "packages/Macaulay2Doc/demo1.m2"; g 2
     ;; packages/Macaulay2Doc/demo1.m2:8:12:(3):[2]: error: division by zero
-    ;;  (1                                            1)   (2      2)   (3      3)
-    ("\\(?:\\(?1:[^\\[{:\n\r\s]+\\)\\|\"\\(?1:.+\\)\"\\):\\([0-9]+\\):\\([0-9]+\\):([0-9]+):\\[[0-9]+\\]"
+    ;;  (1                                             1)   (2      2)   (3      3)
+    ("\\(?:\\(?1:[[:alnum:]/._-]+\\)\\|\"\\(?1:.+\\)\"\\):\\([0-9]+\\):\\([0-9]+\\):([0-9]+):\\[[0-9]+\\]"
```

Comment:
Why not just remove the `-` here? Compatibility with POSIX is in the title, so shouldn't we just remove this?

Reply:
We definitely need `-` in the middle of filenames!  I count 910 .m2 files in the main repository with at least one hyphen.  :)

I just pushed a new version of this commit that doesn't allow hyphens as the first character but does after that.  I think that should fix it.